addon/server.js

```diff
@@ -106,45 +107,59 @@ function extractRouteArguments(args) {
 export default class Server {
 
   constructor(options = {}) {
-    this.environment = options.environment || 'development';
-    this.options = options;
-    this.timing = 400;
-    this.namespace = '';
-    this.urlPrefix = '';
+    this.config(options);
+  }
+
+  config(config = {}) {
+    Ember.assert(!this.environment || this.environment === config.environment,
+    'you can not modify the environment of ember-cli-mirage once server is created');
+    this.environment = config.environment || 'development';
+
+    this.options = config;
+
+    this.timing = this.timing || config.timing || 400;
+    this.namespace = this.namespace || config.namespace || '';
+    this.urlPrefix = this.urlPrefix || config.urlPrefix || '';
 
     this._defineRouteHandlerHelpers();
 
-    this.db = new Db();
-    this.schema = new Schema(this.db, options.models);
-    this.serializerOrRegistry = new SerializerRegistry(this.schema, options.serializers);
+    this.db = this.db || new Db();
+    if (this.schema) {
+      this.schema.registerModels(config.models);
+      this.serializerOrRegistry.schema = this.schema;
+      this.serializerOrRegistry._serializerMap = config.serializers || {};
```

Comment:
instead, let's add an idempotent `registerSerializers` method to the `SerializerRegistry` class that takes in a hash of serializers and augments its registry